src/librustc_mir/borrow_check/nll/renumber.rs

```diff
@@ -62,6 +63,18 @@ impl<'a, 'tcx> MutVisitor<'tcx> for NLLVisitor<'a, 'tcx> {
         debug!("visit_ty: ty={:?}", ty);
     }
 
+    fn process_projection_elem(
+        &mut self,
+        elem: &'b PlaceElem<'tcx>,
+    ) -> Cow<'b, PlaceElem<'tcx>> {
```

Comment:
At this point you can just make this an `Option<PlaceElem<'tcx>>` without any loss in usability, right?

Reply:
Right.